### PR TITLE
Spin off hms as selectable time format

### PR DESCRIPTION
**Pull request by melyux**
The bare minimum changes to do this as talked about in that other PR. Mostly updating `datetime.secondsToClockDuration()` usages. Updated tests

The name of the hms option is up for debate. "Literal" is the stand-in

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9924)

### CODE REVIEW

**Comment by NiLuJe** [comment]
> "Literal" is the stand-in

How about... hear me... "hms"? :D

**Comment by NiLuJe** [comment]
Looks fine at a quick glance (all the changes from modern seem to make sense, AFAICT).

**Comment by poire-z** [comment]
Good to go ?

**Comment by poire-z** [comment]
So, all good ? I trust @melyux about the code changes and expected display results - @Frenzie : ok with the translation tweaks ?

**Comment by Frenzie** [comment]
What was literal again? I recall wanting to think about something better.

**Comment by melyux** [comment]
I think one of us suggested it in the other discussion, which may or may not have been a joke ("littoral" was it?)

My first thought "hms" is not very translation friendly, even tho seems to be what people call it if they ever call it anything. Doesn't matter that much since we give an example next to it

**Comment by poire-z** [comment]
that must have been me, `littéral` in french translates to `literal` in english :/
So, that's what I meant, literal, as plain, obvious, reading as written.

**Comment by Frenzie** [comment]
> My first thought "hms" is not very translation friendly, even tho seems to be what people call it if they ever call it anything. Doesn't matter that much since we give an example next to it

Why do you say it's not?

**Comment by melyux** [comment]
I guess it could be. Do you prefer "Letters" or "HMS"?

**Comment by Frenzie** [comment]
Is there any particular reason to try to avoid hours minutes seconds?

**Comment by melyux** [comment]
Seems out of the pithy theme of "Classic" and "Modern" :)

**Comment by poire-z** [comment]
So, all good (bis) ?

**Comment by Frenzie** [comment]
Well, there's a lack of screenshots and I don't really want to fetch it, so I have no idea one way or the other.

**Comment by melyux** [comment]
Where did we land on the exact wording instead of "Literal"?

**Comment by Frenzie** [comment]
Please provide some screenshots to figure that out. ;-)

**Comment by NiLuJe** [comment]
Gentle ping ;).

**Comment by Frenzie** [comment]
I guess this would be the relevant screenshot? 


As I seem to have said previously, I'd probably just call that HMS with a translator's note (or Letters). But I'd have no idea what literal means if someone said it.

**Comment by poire-z** [comment]
> But I'd have no idea what literal means if someone said it.

I said it, but I won't stand for it :):
> that must have been me, `littéral` in french translates to `literal` in english :/
> So, that's what I meant, literal, as plain, obvious, reading as written.

**Comment by Frenzie** [comment]
> as written

"Written" might be a decent alternative.

**Comment by NiLuJe** [comment]
"Spelled out"?

**Comment by Frenzie** [comment]
That makes me think it's currently twenty-two forty-seven o'clock.

**Comment by NiLuJe** [comment]
True enough ;o).

**Comment by melyux** [comment]
I like "Written" too

**Comment by Frenzie** [comment]
`Letters` and `Written` are both fine by me. Letters then?

**Comment by melyux** [comment]
Pushed

**Review by Frenzie — APPROVED** [review]

File: plugins/statistics.koplugin/main.lua [hunk 1]
```diff
@@ -1872,7 +1872,7 @@ function ReaderStatistics:getDatesFromAll(sdays, ptype, book_mode)
             local stop_month = os.time{year=year_end, month=month_end, day=1, hour=0, min=0 }
             table.insert(results, {
                 date_text,
-                T(N_("%1 (%2 page)", "%1 (%2 pages)", tonumber(result_book[2][i])), datetime.secondsToClockDuration(user_duration_format, tonumber(result_book[3][i]), false, true), tonumber(result_book[2][i])),
+                T(N_("%1 (%2 page)", "%1 (%2 pages)", tonumber(result_book[2][i])), datetime.secondsToClockDuration(user_duration_format, tonumber(result_book[3][i]), false), tonumber(result_book[2][i])),
```
Comment (Frenzie) [inline plugins/statistics.koplugin/main.lua:1875]: What exactly is this saying? Is the first meant to be `%1 (1 page)`?

Comment (melyux) [reply plugins/statistics.koplugin/main.lua:1875]: I think a mistake from when the workings of `gettext` were a mystery, a mystery compounded by the order of the variables, left over from the statistics PR. It should say this, I think:

```lua
T(N_("%2 (1 page)", "%2 (%1 pages)", tonumber(result_book[2][i])), datetime.secondsToClockDuration(user_duration_format, tonumber(result_book[3][i]), false), tonumber(result_book[2][i])),
```

The mismatch between `tonumber(result_book[2][I])` in the N_() and `tonumber(result_book[3][I])` was there before even that, still not sure why they were different but left it alone

Comment (Frenzie) [reply plugins/statistics.koplugin/main.lua:1875]: Would you mind fixing it in a mini-PR if your mind's still in tune with this code?

Comment (melyux) [reply plugins/statistics.koplugin/main.lua:1875]: No problem, but as a sanity check, is the corrected line I posted actually correct?

Comment (Frenzie) [reply plugins/statistics.koplugin/main.lua:1875]: Not entirely, always write it sensibly, i.e., it should look like `"%1 (1 page)", "%1 (%2 pages)"`.

It's the code that should adapt, not what's given to the translators. ^_^

Comment (melyux) [reply plugins/statistics.koplugin/main.lua:1875]: I always always forget that the N_() does nothing except choose one of the strings depending on the number given in the second arg

**Comment by poire-z** [comment]
Btw, wouldn't you prefer your letters'ed duration to get a little spacing ?
Tried it, and `1d23h38m` may feel a bit crowded.
I just noticed that in some private patched feature of mine, I output it as `1d 23h 38m` with regular spaces.
Could look in-between these with thin/hair spaces ? Unless the usage is to see all that stuck ? (And may be it's a french thing to add space everywhere like before a question mark :)

**Comment by melyux** [comment]
@poire-z I made the exact same patch for myself, with the added spaces! But didn't think you guys would like it. I think it looks better with the spaces, no hairline required. I'm down to do a mini PR to add the space if everyone else is fine with it too

**Comment by NiLuJe** [comment]
No objection here ;).

**Pull request merged**